pallets/parachain-staking/src/benchmarks.rs

```diff
@@ -1618,6 +1618,116 @@ benchmarks! {
 	verify {
 	}
 
+	pay_one_collator_reward_best {
+		// x controls number of delegations
+		let x in 0..(
+			T::MaxTopDelegationsPerCandidate::get()
+			+ T::MaxBottomDelegationsPerCandidate::get() - 1
+		);
+		// y controls the number of auto-compounding delegations
+		let y in 0..(
+			T::MaxTopDelegationsPerCandidate::get()
+			+ T::MaxBottomDelegationsPerCandidate::get() - 1
+		);
+		// z is the number of scheduled requests per collator
+		let z in 0..(
+			T::MaxTopDelegationsPerCandidate::get()
+			+ T::MaxBottomDelegationsPerCandidate::get() - 1
+		);
+
+		use crate::{
+			DelayedPayout, DelayedPayouts, AtStake, CollatorSnapshot, BondWithAutoCompound, Points,
+			AwardedPts,
+		};
+
+		let mut seed = Seed::new();
+		let prime_candidate = create_account::<T>(
+			"collator",
+			seed.take(),
+			AccountBalance::MinCandidateStake,
+			AccountAction::JoinCandidates{ amount: Amount::All, candidate_count: 1u32 },
+		)?;
+
+		let mut delegations = Vec::new();
+		let mut col_del_count = 0u32;
+		let initial_delegator_balance = T::MinDelegation::get() + 100u32.into();
+		for i in 0..x {
+			let auto_compound = if i+1 <= y { Percent::from_percent(100) } else { Percent::from_percent(0) };
```

Comment:
Conditions like this can lead to skewed linear regression results in cases where `y > x` since it will artificially limit the impact of `y`. Are we sure that's not happening here?

Reply:
To be clear, I'm assuming that `Percent::from_percent(0)` implies some work isn't being done, maybe I'm wrong about that...

Reply:
Context: I'm thinking of the problem solved here: https://github.com/moonbeam-foundation/moonbeam/pull/2336

Reply:
Yeah this is the same issue in our other benchmarks. The slope is however calculated correctly, we visualized them once from this random python script Shawn posted. It's unfortuataely a limitation that merely wastes cycles.

Reply:
> Context: I'm thinking of the problem solved here: #2336

This issue was resolved by not testing for `BottomDelegations` as that caused the slope detector to go haywire.